src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -255,16 +255,19 @@ public void deleteIndexDirectorySafe(Index index, long lockTimeoutMS, @IndexSett
      * @return the {@link ShardLock} instances for this index.
      * @throws IOException if an IOException occurs.
      */
-    public List<ShardLock> lockAllForIndex(Index index, long lockTimeoutMS) throws IOException {
-        Set<ShardId> allShardIds = findAllShardIds(index);
-        logger.trace("locking all shards for index {} - [{}]", index, allShardIds);
-        List<ShardLock> allLocks = new ArrayList<>(allShardIds.size());
+    public List<ShardLock> lockAllForIndex(Index index, @IndexSettings Settings settings, long lockTimeoutMS) throws IOException {
+        final Integer numShards = settings.getAsInt(IndexMetaData.SETTING_NUMBER_OF_SHARDS, null);
+        if (numShards == null || numShards <= 0) {
+            throw new IllegalArgumentException("settings must contain a non-null > 0 number of shards");
+        }
+        logger.trace("locking all shards for index {} - [{}]", index, numShards);
```

Comment:
can we use the index name first in the log, like `[{}] locking all shards, num_shards [{}], index, numShards`